src/fmureceiver.cpp

```diff
@@ -0,0 +1,307 @@
+    }
+    else
+    {
+        callbacks_.log_level = jm_log_level_debug;
+    }
+    callbacks_.context = 0;
+
+    context_ = fmi_import_allocate_context(&callbacks_);
+    fmi_version_enu_t version = fmi_import_get_fmi_version(context_, FMUPath_.c_str(), tmpPath_.c_str());
+
+    if (version != fmi_version_2_0_enu)
+    {
+        qDebug() << "The code only supports version 2.0";
+        return false;
+    }
+    return true;
+}
+
+bool
+FMUReceiver::importFMU()
+{
+    // TODO: config or use default values?
+    callBackFunctions_.logger = fmi2_log_forwarding;
+    callBackFunctions_.allocateMemory = calloc;
+    callBackFunctions_.freeMemory = free;
+    callBackFunctions_.componentEnvironment = fmu_;
+
+    jmStatus_ = fmi2_import_create_dllfmu(fmu_, fmi2_fmu_kind_cs, &callBackFunctions_);
+
+    if (jmStatus_ == jm_status_error)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool
+FMUReceiver::initializeFMU()
+{
+    // TODO: config or use default values?
+    // TODO: alternative solution for start values & verify parameters for setup and initiliazation functions
+    // start values
+    fmi2_string_t instanceName = "Test CS model instance";
+
+    fmi2_string_t fmuLocation = "";
+    fmi2_boolean_t visible = fmi2_false;
+    fmi2_real_t relativeTol = 1e-4;
+
+    tStart_ = 0;
+    tCurrent_ = tStart_;
+    fmi2_boolean_t StopTimeDefined = fmi2_false;
+
+    // Do we need it?
+    fmi2_string_t fmuGUID;
+    fmuGUID = fmi2_import_get_GUID(fmu_);
+
+    jmStatus_ = fmi2_import_instantiate(fmu_, instanceName, fmi2_cosimulation, fmuLocation, visible);
+    fmiStatus_ = fmi2_import_setup_experiment(fmu_, fmi2_true, relativeTol, tStart_, StopTimeDefined, tEnd_);
+    fmiStatus_ = fmi2_import_enter_initialization_mode(fmu_);
+    fmiStatus_ = fmi2_import_exit_initialization_mode(fmu_);
+
+    vr_[FMI_INTEGER_SENSORDATA_IN_BASELO_IDX] = FMI_INTEGER_SENSORDATA_IN_BASELO_IDX;
+    vr_[FMI_INTEGER_SENSORDATA_IN_BASEHI_IDX] = FMI_INTEGER_SENSORDATA_IN_BASEHI_IDX;
+    vr_[FMI_INTEGER_SENSORDATA_IN_SIZE_IDX] = FMI_INTEGER_SENSORDATA_IN_SIZE_IDX;
+    vr_[FMI_INTEGER_SENSORDATA_OUT_BASELO_IDX] = FMI_INTEGER_SENSORDATA_OUT_BASELO_IDX;
+    vr_[FMI_INTEGER_SENSORDATA_OUT_BASEHI_IDX] = FMI_INTEGER_SENSORDATA_OUT_BASEHI_IDX;
+    vr_[FMI_INTEGER_SENSORDATA_OUT_SIZE_IDX] = FMI_INTEGER_SENSORDATA_OUT_SIZE_IDX;
+
+    integerVars_[FMI_INTEGER_SENSORDATA_IN_BASELO_IDX] = 0;
+    integerVars_[FMI_INTEGER_SENSORDATA_IN_BASEHI_IDX] = 0;
+    integerVars_[FMI_INTEGER_SENSORDATA_IN_SIZE_IDX] = 0;
+
+    integerVars_[FMI_INTEGER_SENSORDATA_OUT_BASELO_IDX] = 0;
+    integerVars_[FMI_INTEGER_SENSORDATA_OUT_BASEHI_IDX] = 0;
+    integerVars_[FMI_INTEGER_SENSORDATA_OUT_SIZE_IDX] = 0;
+
+    /* Boolean Variables defined in the OSMPCNetworkProxy.h */
+    //    #define FMI_BOOLEAN_DUMMY_IDX 0
+    //    #define FMI_BOOLEAN_SENDER_IDX 1
+    //    #define FMI_BOOLEAN_RECEIVER_IDX 2
+    fmi2_boolean_t booleanVars_[3];
```

Comment:
Dito for the booleans, but here OSMP does not guarantee that they exist at all, so this should be user configurable...